communicator/ssh/provisioner.go

```diff
@@ -127,11 +128,9 @@ func decodeConnInfo(v cty.Value) (*connectionInfo, error) {
 		case "bastion_host_key":
 			connInfo.BastionHostKey = v.AsString()
 		case "bastion_port":
-			p, err := strconv.Atoi(v.AsString())
-			if err != nil {
+			if err := gocty.FromCtyValue(v, &connInfo.BastionPort); err != nil {
```

Comment:
An interesting consequence of this approach is that if we were to change the type of this `BastionPort` field to be `uint16` then I believe `gocty` will also automatically produce a range error if the value is outside of the expected range of port numbers.

I'm not meaning to say "you should do this" here, but more just an FYI in case you want to give it a try and see what the implications are. Fine to leave that for another layer to validate too, though.

Reply:
That's a neat idea, I hadn't considered it! Thanks.